### PR TITLE
Added additional pose offset for the performer detector

### PR DESCRIPTION
**Pull request by nkoenig**
It's currently not possible to set the pose of an `<include>`ed performer detector relative to the parent model.

For example, consider this parent model:
```
<sdf version="1.6">
    <model name="cave_starting_area">
        <static>true</static>
        <include>
          <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Staging Exit Detector</uri>
        </include>
....
  </model>
</sdf>
```

And the included `Cave Staging Exit Detector`.
```
<sdf version="1.6">
    <model name="cave_staging_exit_detector">
        <static>true</static>
        <pose>13.5 0 0 0 0 0</pose>
       <plugin filename="libignition-gazebo-performer-detector-system.so" name="ignition::gazebo::systems::PerformerDetector">
            <topic>/subt_performer_detector</topic>
            <geometry>
                <box>
                    <size>4 30 20</size>
                </box>
            </geometry>
        </plugin>
    </model>
```

The `<pose>` in `Cave Staging Exit Detector` is not used by the plugin, because of the `<include>` process. You also cannot set the `<pose>` in the `<include>` statement for the same reason.

This means the plugin is always located at `0 0 0 0 0 0` in the parent model.

This PR adds an optional `<pose>` to the plugin to address this situation.

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
    JointStatePublisher.
     * [Pull Request 222](https://github.com/ignitionrobotics/ign-gazebo/pull/222)
 
+1. Added an additional pose offset for the performer detector plugin.
+    * [Pull Request 235](https://github.com/ignitionrobotics/ign-gazebo/pull/235)
```
Comment (iche033) [inline Changelog.md:10]:
```suggestion
    * [Pull Request 236](https://github.com/ignitionrobotics/ign-gazebo/pull/236)
```

File: src/systems/performer_detector/PerformerDetector.hh [hunk 2]
```diff
@@ -67,6 +67,8 @@ namespace systems
   /// `<geometry>`: Detection region. Currently, only the `<box>` geometry is
   /// supported. The position of the geometry is derived from the pose of the
   /// containing model.
+  /// `<pose>`: Additional pose offset. This pose is added to the parent
+  /// model pose when computing the detection region.
```
Comment (iche033) [inline src/systems/performer_detector/PerformerDetector.hh:71]: mention that the rotational component of pose is not used?

Comment (nkoenig) [reply src/systems/performer_detector/PerformerDetector.hh]: Added: 359759f00c2ce098250f14bd032c7acc37398d32

File: src/systems/performer_detector/PerformerDetector.cc [hunk 3]
```diff
@@ -109,7 +114,8 @@ void PerformerDetector::PostUpdate(
 
   // Double negative because AxisAlignedBox does not currently have operator+
   // that takes a position
-  auto region = this->detectorGeometry - (-modelPose.Pos());
+  auto region = this->detectorGeometry -
+    (-(modelPose.Pos() + this->poseOffset.Pos()));
```
Comment (azeey) [inline src/systems/performer_detector/PerformerDetector.cc:118]: This implies that the position offset is given in world frame. If we want it to be relative relative to the model, I think this should be
```suggestion
    (-(modelPose.Pos() + modelPose.Rot() * this->poseOffset.Pos()));
```

File: src/systems/performer_detector/PerformerDetector.hh [hunk 4]
```diff
@@ -67,6 +67,9 @@ namespace systems
   /// `<geometry>`: Detection region. Currently, only the `<box>` geometry is
   /// supported. The position of the geometry is derived from the pose of the
   /// containing model.
+  /// `<pose>`: Additional pose offset. This pose is added to the parent
```
Comment (azeey) [inline src/systems/performer_detector/PerformerDetector.hh:70]: Could you add the frame relative to which this pose is given?

File: src/systems/performer_detector/PerformerDetector.cc (hunk 3 above)
Comment (nkoenig) [reply src/systems/performer_detector/PerformerDetector.cc]: Thanks: e34d3bda28d79ebbbc82a8d473e0452b2069971f

File: src/systems/performer_detector/PerformerDetector.hh (hunk 4 above)
Comment (nkoenig) [reply src/systems/performer_detector/PerformerDetector.hh]: Add: e34d3bda28d79ebbbc82a8d473e0452b2069971f

**Comment by nkoenig** [comment]
> LGTM! My only concern with this PR is that the nesting example you gave, even with this PR, leads to unexpected results. i.e, with parent model:
> 
> ```
> <sdf version="1.6">
>     <model name="cave_starting_area">
>         <static>true</static>
>         <include>
>           <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Staging Exit Detector</uri>
>            <pose>0 0 10 0 0 0</pose>
>         </include>
> ....
>   </model>
> </sdf>
> ```
> 
> And the included `Cave Staging Exit Detector`.
> 
> ```
> <sdf version="1.6">
>     <model name="cave_staging_exit_detector">
>         <static>true</static>
>         <pose>13.5 0 0 0 0 0</pose>
>        <plugin filename="libignition-gazebo-performer-detector-system.so" name="ignition::gazebo::systems::PerformerDetector">
>             <topic>/subt_performer_detector</topic>
>             <geometry>
>                 <box>
>                     <size>4 30 20</size>
>                 </box>
>             </geometry>
>         </plugin>
>     </model>
> ```
> 
> The expected pose of the detector's box in world frame is `13.5 0 10 0 0 0`, but the result will still be `0 0 0 0 0 0`.

Yup, thi

> LGTM! My only concern with this PR is that the nesting example you gave, even with this PR, leads to unexpected results. i.e, with parent model:
> 
> ```
> <sdf version="1.6">
>     <model name="cave_starting_area">
>         <static>true</static>
>         <include>
>           <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Staging Exit Detector</uri>
>            <pose>0 0 10 0 0 0</pose>
>         </include>
> ....
>   </model>
> </sdf>
> ```
> 
> And the included `Cave Staging Exit Detector`.
> 
> ```
> <sdf version="1.6">
>     <model name="cave_staging_exit_detector">
>         <static>true</static>
>         <pose>13.5 0 0 0 0 0</pose>
>        <plugin filename="libignition-gazebo-performer-detector-system.so" name="ignition::gazebo::systems::PerformerDetector">
>             <topic>/subt_performer_detector</topic>
>             <geometry>
>                 <box>
>                     <size>4 30 20</size>
>                 </box>
>             </geometry>
>         </plugin>
>     </model>
> ```
> 
> The expected pose of the detector's box in world frame is `13.5 0 10 0 0 0`, but the result will still be `0 0 0 0 0 0`.

Yup, that is not intuitive, but I don't know how to fix that right now.

**Pull request merged**